contracts/AaveMiddleware.sol

```diff
@@ -0,0 +1,180 @@
+import "./Iweth.sol";
+
+/// @title A contract for managing Aave positions. Acts as a middleware
+/// @author Aditya Choudhary
+/// @notice Does 4 basic operations of Deposit, Withdraw, Borrow and Repay for both ERC20 and ETH.
+///@dev Maintains User's Balance on contract to make sure withdrawal is not made for amount greated than deposited.
+
+contract AaveMiddleware {
+  using SafeERC20 for IERC20;
+
+  function getLendingPoolAddress() public view returns (address) {
+    return
+      ILendingPoolAddressesProvider(
+        address(0xB53C1a33016B2DC2fF3653530bfF1848a515c8c5)
+      ).getLendingPool(); //Mainnet
+  }
+
+  function depositToken(uint256 _amount, address _tokenAddr) external {
+    address TokenAddress = _tokenAddr; //Token Address on Mainnet. Re-initiated in all functions to avoid gas cost associated with state varaibles.
+    IERC20 erc20Token = IERC20(TokenAddress);
+    address _LendingPoolAddress = getLendingPoolAddress();
+    ILendingPool LendingPool = ILendingPool(_LendingPoolAddress);
+
+    erc20Token.safeTransferFrom(msg.sender, address(this), _amount); //Assuming the User has already Approved this amount using approve() function
+
+    erc20Token.safeApprove(_LendingPoolAddress, _amount);
+    LendingPool.deposit(TokenAddress, _amount, address(this), 0); //The referral program is currently inactive and you can pass 0 as thereferralCode.
+    LendingPool.setUserUseReserveAsCollateral(_tokenAddr, true); //Not needed
+  }
+
+  function withdrawToken(uint256 _amount, address _tokenAddr) external {
+    address TokenAddress = _tokenAddr; //Token Address on Mainnet
+    ILendingPool LendingPool = ILendingPool(address(getLendingPoolAddress()));
+    LendingPool.withdraw(TokenAddress, _amount, msg.sender); //Amount Withdrawn to the user
+  }
+
+  function borrowToken(uint256 _amount, address _tokenAddr) external {
+    address TokenAddress = _tokenAddr; //Token Address on Mainnet
+    IERC20 erc20Token = IERC20(TokenAddress);
+    address _LendingPoolAddress = getLendingPoolAddress();
+    ILendingPool LendingPool = ILendingPool(_LendingPoolAddress);
+    LendingPool.borrow(TokenAddress, _amount, 2, 0, address(this)); //Amount borrowed. Will be received by the contract
+    erc20Token.safeTransfer(msg.sender, _amount); //Transferring the borrowed funds to the user
+  }
+
+  function repayToken(uint256 _amount, address _tokenAddr) external {
+    address TokenAddress = _tokenAddr; //Token Address on Mainnet
+    IERC20 erc20Token = IERC20(TokenAddress);
+    address _LendingPoolAddress = getLendingPoolAddress();
+    ILendingPool LendingPool = ILendingPool(_LendingPoolAddress);
+    erc20Token.safeTransferFrom(msg.sender, address(this), _amount); //Assuming the User has already Approved this amount using approve() function
+    erc20Token.safeApprove(_LendingPoolAddress, _amount);
+    LendingPool.repay(TokenAddress, _amount, 1, address(this)); //Amount borrowed
```

Comment:
Why `rateMode = 1` in here, while in borrow you used `rateMode = 2`?

Reply:
Oh yeah my bad. Thanks

Reply:
Corrected